pkg/k8sutils/job.go

```diff
@@ -78,10 +78,13 @@ func (k8s *k8sImpl) CreateK8sJob(jobName string, action *config.Action, task con
 		return fmt.Errorf("could not prepare env for job %v: %v", jobName, err.Error())
 	}
 
+	if namespace != "" {
+		namespace = k8s.namespace
```

Comment:
I would suggest deleting the namespace from the k8s struct and just use the namespace passed in as parameter of the function. Because with the current approach it can be error-prone to overwrite the namespace in each function call (see functions `DeleteK8sJob` and `GetLogsOfPod` where k8s.namespace is used but not overwritten yet => would probably result in an error)

The task loop can then decide if to pass in the default namespace or the task specific namespace to each function call.

Reply:
refactored